src/hooks/useCKBNode.tsx

```diff
@@ -24,15 +51,80 @@ interface CKBNodeProviderProps {
 }
 
 export const CKBNodeProvider = ({ children, defaultEndpoint }: PropsWithChildren<CKBNodeProviderProps>) => {
-  const nodeService = new NodeService(defaultEndpoint)
+  const { t } = useTranslation()
+  const connectedEndpoint = localStorage.getItem(NODE_CONNECTED_ENDPOINT) ?? defaultEndpoint
+  const [nodeService, setNodeService] = useState(new NodeService(connectedEndpoint))
+  const [nodes, setNodes] = useState<CKBNode[]>(loadEndpoints())
   const [isActivated, _setIsActivated] = useState(localStorage.getItem(NODE_CONNECT_MODE_KEY) === 'true')
 
   const setIsActivated = (value: boolean) => {
     localStorage.setItem(NODE_CONNECT_MODE_KEY, value.toString())
     _setIsActivated(value)
   }
 
+  const saveNodes = (nodes: CKBNode[]) => {
+    setNodes(nodes)
+    saveEndpoints(nodes)
+  }
+
+  const addNode = (node: CKBNode) => {
+    const existed = nodes.some(n => n.url === node.url)
+    if (existed) return false
+
+    saveNodes([...nodes, node])
+    return true
+  }
+
+  const removeNode = (url: string) => {
+    saveNodes(nodes.filter(node => node.url !== url))
+  }
```

Comment:
1. It looks like `nodes` can be `Map<url: string, alias: string>` which is more intuitive
2. Why is `url` unique but `alias` can be duplicated? Generally, we will see a list in which items have unique index/alias